src/components/dropdown-menu/dropdown-menu.jsx

```diff
@@ -0,0 +1,205 @@
+	} );
+
+	const toggleMenu = () => setIsOpen( ( prev ) => ! prev );
+
+	const handleClose = () => setIsOpen( false );
+
+	return (
+		<DropdownMenuContext.Provider value={ { handleClose } }>
+			<div className={ cn( 'relative inline-block', className ) }>
+				<div
+					ref={ refs.setReference }
+					onClick={ toggleMenu }
+					role="button"
+					tabIndex={ 0 }
+					{ ...getReferenceProps() }
+					className="cursor-pointer"
+				>
+					{ React.Children.map( children, ( child ) => {
+						if (
+							child.type?.displayName === 'DropdownMenu.Trigger'
+						) {
+							return child;
+						}
+						return null;
+					} ) }
+				</div>
+
+				{ isMounted && (
+					<FloatingPortal
+						id={ dropdownPortalId }
+						root={ dropdownPortalRoot }
+					>
+						<div
+							ref={ refs.setFloating }
+							style={ {
+								...floatingStyles,
+								...styles,
+							} }
+							{ ...getFloatingProps() }
+						>
+							{ React.Children.map( children, ( child ) => {
+								if (
+									child.type?.displayName ===
+									'DropdownMenu.Content'
+								) {
+									return child;
+								}
+								return null;
+							} ) }
+						</div>
+					</FloatingPortal>
+				) }
+			</div>
+		</DropdownMenuContext.Provider>
+	);
+};
+
+DropdownMenu.displayName = 'DropdownMenu';
+
+const DropdownMenuTrigger = React.forwardRef( ( { children, className }, ref ) => (
+	<div ref={ ref } role="button" tabIndex={ 0 } className={ className }>
+		{ children }
+	</div>
+) );
+
+DropdownMenuTrigger.displayName = 'DropdownMenu.Trigger';
+
+const DropdownMenuContent = ( { children, className, ...props } ) => {
+	return (
+		<div
+			className={ cn(
+				'border border-solid border-border-subtle rounded-md shadow-lg overflow-hidden',
+				className
+			) }
+		>
+			<Menu { ...props }>{ children }</Menu>
+		</div>
+	);
+};
+DropdownMenuContent.displayName = 'DropdownMenu.Content';
+
+const DropdownMenuList = ( { children, ...props } ) => {
+	const { handleClose } = useDropdownMenuContext();
+
+	return (
+		<Menu.List { ...props }>
+			{ React.Children.map( children, ( child ) => {
+				if ( React.isValidElement( child ) ) {
+					return React.cloneElement( child, {
+						onClick: callAll( child.props?.onClick, handleClose ),
+					} );
+				}
+				return child;
+			} ) }
+		</Menu.List>
+	);
+};
+
+DropdownMenuList.displayName = 'DropdownMenu.List';
+
+const DropdownMenuItem = ( { children, as: Tag = Fragment, ...props } ) => {
+	const { handleClose } = useDropdownMenuContext();
+
+	if ( ! isValidElement( children ) || ! children ) {
+		return (
+			<Menu.Item { ...props } onClick={ callAll( handleClose, props.onClick ) }>
+				{ children }
+			</Menu.Item>
+		);
+	}
+
+	if ( Tag === Fragment ) {
+		return cloneElement( children, {
+			onClick: callAll( children.props?.onClick, handleClose ),
+		} );
+	}
+
+	return (
+		<Tag { ...props } onClick={ callAll( props.onClick, handleClose ) }>
+			{ children }
+		</Tag>
+	);
+};
```

Comment:
Here, every menu item will have children. These can be text, elements, or other components. If the children are not valid, we need to return null. By default, we will use the `Menu.Item` component. If developers don't want to use the predefined style, they can pass `Fragment` and use other components as children. `cloneElement` will handle the attaching close handler.

```suggestion
const DropdownMenuItem = ( { children, as: Tag = Menu.Item , ...props } ) => {
	const { handleClose } = useDropdownMenuContext();

	if ( ! isValidElement( children ) || ! children ) {
		return null;
	}

	if ( Tag === Fragment ) {
		return cloneElement( children, {
			onClick: callAll( children.props?.onClick, handleClose ),
		} );
	}

	return (
		<Tag { ...props } onClick={ callAll( props.onClick, handleClose ) }>
			{ children }
		</Tag>
	);
};
```